src/package-operations.ts

```diff
@@ -250,57 +266,102 @@ function getUpdatedDependencyField(
   return manifest as Pick<PackageManifest, T>;
 }
 
 /**
- * Validates a manifest by ensuring that the given required fields are present
- * and properly formatted.
+ * Validates a manifest by ensuring that the given fields are properly formatted
+ * if present. Fields that are required by the `PackageManifest` interface must be
+ * present if specified.
  *
+ * @see PackageManifest - For fields that must be present if specified.
  * @param manifest - The manifest to validate.
  * @param manifestDirPath - The path to the directory containing the
  * package.json file.
- * @param requiredFields - The manifest fields that will be required during
- * validation.
+ * @param fieldsToValidate - The manifest fields that will be validated.
  */
 function validatePackageManifest(
-  manifest: Record<string, unknown>,
+  manifest: Partial<PackageManifest>,
   manifestDirPath: string,
-  requiredFields: (keyof PackageManifest)[] = ['name', 'version'],
+  fieldsToValidate: (keyof PackageManifest)[] = [
+    FieldNames.Name,
+    FieldNames.Version,
+  ],
 ): void {
-  if (requiredFields.length === 0) {
+  if (fieldsToValidate.length === 0) {
     return;
   }
+  const _fieldsToValidate = new Set(fieldsToValidate);
 
   // Just for logging purposes
   const legiblePath = manifestDirPath.split('/').splice(-2).join('/');
+  const getErrorMessagePrefix = (fieldName: FieldNames) => {
+    return `${
+      manifest[FieldNames.Name]
+        ? `"${manifest[FieldNames.Name]}" manifest "${fieldName}"`
+        : `"${fieldName}" of manifest in "${legiblePath}"`
+    }`;
+  };
+
+  if (
+    _fieldsToValidate.has(FieldNames.Name) &&
+    !isTruthyString(manifest[FieldNames.Name])
+  ) {
+    throw new Error(
+      `Manifest in "${legiblePath}" does not have a valid "${FieldNames.Name}" field.`,
+    );
+  }
+
+  if (
+    _fieldsToValidate.has(FieldNames.Version) &&
+    !isValidSemver(manifest[FieldNames.Version])
+  ) {
+    throw new Error(
+      `${getErrorMessagePrefix(
+        FieldNames.Version,
+      )} is not a valid SemVer version: ${manifest[FieldNames.Version]}`,
+    );
+  }
 
-  if (requiredFields.includes('name') && !isTruthyString(manifest.name)) {
+  if (
+    _fieldsToValidate.has(FieldNames.Private) &&
+    FieldNames.Private in manifest &&
+    typeof manifest[FieldNames.Private] !== 'boolean'
+  ) {
     throw new Error(
-      `Manifest in "${legiblePath}" does not have a valid "name" field.`,
+      `${getErrorMessagePrefix(
+        FieldNames.Private,
+      )} must be a boolean if present. Received: ${
+        manifest[FieldNames.Private]
+      }`,
     );
   }
 
-  if (requiredFields.includes('version') && !isValidSemver(manifest.version)) {
+  if (
+    _fieldsToValidate.has(FieldNames.Workspaces) &&
+    FieldNames.Workspaces in manifest &&
+    (!Array.isArray(manifest[FieldNames.Workspaces]) ||
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
Huh. Odd that that's needed. I tried replacing the `FieldNames.Workspaces in manifest` above with an explicit `undefined` and `null` check, but this non-null assertion still seems required for some reason... I guess TypeScript doesn't clue into type guards within a conditional expression.